func_test.go

```diff
@@ -92,26 +94,17 @@ func TestNilFuncArgs(t *testing.T) {
 		},
 	} {
 		t.Run(c.name, func(t *testing.T) {
-			checkPanic := func() {
-				r := recover()
-				if c.ok {
-					if r != nil {
-						t.Errorf("expected no panic, got %v", r)
-					}
-				} else {
-					if r == nil {
-						t.Errorf("expected panic")
-					}
-				}
+			var matcher *h.Matcher
+			if c.ok {
+				matcher = h.Panics(h.Nil())
+			} else {
+				matcher = h.Panics(h.HasSubstr("nil"))
```

Comment:
Does this have the same effect as `h.Panics(h.NotNil())`? That's what [the docs seem to recommend](https://pkg.go.dev/github.com/grailbio/testutil/h?tab=doc#Panics).

Reply:
This checks that the string "nil" is in the message, as we expect the panic message to say something about nil-ness.

Reply:
I'll add a comment explaining this.